### DISCUSSION TITLE
Validating the input of a number between 0000 and 9999 with leading zeros

### DISCUSSION THREAD

**Discussion by Paul2701** [primefaces/community#2845] [PrimeFaces] [answered]
Hi guys,
I need to enter a 4-digit number and validate it immediate: only 0000 to 9999 are allowed.
The example here https://www.primefaces.org/showcase/ui/csv/immediate.xhtml?jfwid=8473c for Regex "^[0-9]{4}$" does not work for me.
Thanks for any help in advance.
Paul

**Comment by eugene-lazarev** [comment]
@Paul2701, sorry for disturbing you, but maybe component [OTPInput](https://www.primefaces.org/showcase-ext/sections/inputotp/basicUsage.jsf) fits better for you?

**Comment by Paul2701** [comment]
@eugene-lazarev thanx :-), but no...

**Reply by melloware** [reply]
What about...

```xml
<p:inputMask value="#{maskView.yours}" mask="9999" validateMask="true" />
```

**Reply by Paul2701** [reply]
yes, thanks @melloware! That would be a good alternative, I just want to use this cool feature:
![image](https://github.com/user-attachments/assets/d308c03c-681d-44b3-a265-8db4a6bbfa58)

**Reply by melloware** [reply]
I just tested `<f:validateRegex pattern="^[0-9]{4}$"/>` and its working for me.

![image](https://github.com/user-attachments/assets/6610140c-57d8-4b4f-b9a8-0c37443edc27)

![image](https://github.com/user-attachments/assets/ef467587-6764-4ccd-980d-9b87c6e35ae2)

**Reply by Paul2701** [reply]
@melloware 
strange
I use the same style and this:
`          <div class="position">
            <span class="ui-float-label">
            <p:inputText id="regex" value="#{MyBean.position}"
                         validatorMessage="Value does not match pattern.">
                <f:validateRegex pattern="^[a-zA-Z]+$"/>
                <p:clientValidator event="keyup"/>
            </p:inputText>
            <p:message for="regex"/>
            <h:outputText value="#{MyBean.position}"/>
              <p:outputLabel for="regex" value="#{msg['structure.position']}"/>
            </span>
          </div>
`
In your example it works:
![image](https://github.com/user-attachments/assets/9f78f250-1b18-42f9-b42b-446089812ebc)

and I see green:
![image](https://github.com/user-attachments/assets/84412b13-f162-460d-b1a1-762dd9a72c29)

**Reply by melloware** [reply]
What version of PF are you using?

**Reply by Paul2701** [reply]
14.0.5

**Reply by melloware** [reply, marked as answer]
And you have this enabled in your web.xml?

```xml
    <context-param>
        <param-name>primefaces.CLIENT_SIDE_VALIDATION</param-name>
        <param-value>true</param-value>
    </context-param>
```

**Reply by Paul2701** [reply]
yes, that's it!
Thanks! A cool feature, I think! :-)

**Reply by melloware** [reply]
Yep that moves validation to the browser instead of the server. Its very slick.
